### PR TITLE
Fixed #21080 -- Ignored urls inside static files comments.

### PR DESCRIPTION
**Pull request by n6g7**
Added comment blocks detection to prevent replacement of urls inside
css comments.

### CODE REVIEW

File: tests/staticfiles_tests/project/documents/cached/css/comments.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+/* Single line: @import url("does.not.exist.A.css"); */
+
+/* Multi lines:
+ * @import url("does.not.exist.B.css");
+ */
+
+@font-face {
+  src: url("fonts/font.eot?#iefix") format("embedded-opentype");
+  /* Single line: src: url("does.not.exist.C.eot") format("embedded-opentype"); */
+}
```
Comment (felixxm) [inline tests/staticfiles_tests/project/documents/cached/css/comments.css]: Can we add a test for inline comment? e.g.
```css
@font-face {
    src: /*url("does.not.exist.E.eot")*/ url("fonts/font.eot?#iefix") format("embedded-opentype");
}
```
and also for two comments in a line, e.g.
```css
@font-face {
    src: /*comment 1*/ url("fonts/font.eot?#iefix") /*comment 2*/ format("embedded-opentype");
}
```

**Comment by carltongibson** [comment]
Hi @n6g7.

This is the sort of thing where I'd normally weep, but first glance it doesn't look too bad. 🙂 

I'll have a play and see if I can break it. 
(Any thoughts @claudep?)

File: django/contrib/staticfiles/storage.py [hunk 2]
```diff
@@ -151,7 +151,24 @@ def url(self, name, force=False):
...
         return self._url(self.stored_name, name, force)
 
-    def url_converter(self, name, hashed_files, template=None):
+    def get_comment_blocks(self, content):
+        """
+        Return a list of (start, end) tuples for each comment block.
+        """
+        return [
+            (match.start(), match.end())
+            for match in re.finditer(r"\/\*.*?\*\/", content, flags=re.DOTALL)
```
Comment (felixxm) [inline django/contrib/staticfiles/storage.py]: I think we should use a bit more complicated regular expression `\/\*[^*]*\*+([^/*][^*]*\*+)*\/` (see ["lexical scanner"](https://www.w3.org/TR/CSS22/grammar.html#scanner) and ["consume comments"](https://drafts.csswg.org/css-syntax-3/#consume-comment)).

**Comment by claudep** [comment]
If we admit this is a "best effort" method, then it's still an improvement IMHO. But parsing syntax with regex is bound to fail at some corner cases, so we must be prepared to that. Unfortunately don't think there is a real stable CSS parser in the Python world currently, at least not one I would be comfortable to add as an official dependency.

File: django/contrib/staticfiles/storage.py (hunk 2 above)
Comment (felixxm) [reply django/contrib/staticfiles/storage.py]: ... also we should take into account only comments that contain `url` or `import`.

**Comment by tmszi** [comment]
I write own solution [PR](https://github.com/django/django/pull/12107)

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**